packages/frontend/src/components/dialogs/ProxyConfiguration/index.tsx

```diff
@@ -0,0 +1,380 @@
+  props: DialogProps & { accountId: number; configured: boolean }
+) {
+  const tx = useTranslationFunction()
+
+  // used in  new proxy form
+  const [newProxyUrl, setNewProxyUrl] = useState('')
+
+  const [showNewProxyForm, setShowNewProxyForm] = useState(false)
+
+  // updated on connectivity change
+  const [connectivityStatus, setConnectivityStatus] = useState(
+    C.DC_CONNECTIVITY_NOT_CONNECTED
+  )
+
+  // configured means the account is already configured
+  // which is needed to decide if we show the connectivity status
+  const { accountId, configured, onClose } = props
+
+  const openAlertDialog = useAlertDialog()
+
+  const [proxyState, setProxyState] = useState<ProxyStateType>({
+    enabled: false,
+    proxies: [],
+    activeProxy: null,
+    updateSettings: false,
+  })
+
+  // convenience function to update the proxy state
+  // called only after user actions (so updateSettings is set to true)
+  const updateProxyState = useCallback(
+    (updates: Partial<typeof proxyState>) =>
+      setProxyState(prev => ({ ...prev, ...updates, updateSettings: true })),
+    [setProxyState]
+  )
+
+  useEffect(() => {
+    const loadSettings = async () => {
+      try {
+        const { proxy_enabled, proxy_url } =
+          await BackendRemote.rpc.batchGetConfig(accountId, [
+            'proxy_enabled',
+            'proxy_url',
+          ])
+        if (proxy_enabled !== undefined) {
+          const enabled = proxy_enabled === Proxy.ENABLED
+          let proxies: string[] = []
+          let activeProxy = null
+          if (proxy_url && proxy_url.length > 0) {
+            // split proxy_url by new line
+            // and remove empty lines from possible previous settings
+            const proxyLines = proxy_url.split(/\n/).filter(s => !!s)
+            proxies = proxyLines
+            activeProxy = enabled ? proxyLines[0] : null
+            setShowNewProxyForm(false)
+          } else {
+            setShowNewProxyForm(true)
+          }
+          setProxyState(prev => ({
+            ...prev,
+            enabled,
+            proxies,
+            activeProxy,
+          }))
+        }
+      } catch (error: any) {
+        log.error('failed to load proxy settings', error)
+        openAlertDialog({
+          message: `${tx('error')} ${error.message || error.toString()}`,
+        })
+      }
+    }
+    loadSettings()
+  }, [accountId, openAlertDialog, tx])
+
+  const changeProxyEnable = (enableProxy: boolean) => {
+    let activeProxy = null
+    // if proxy is disabled, set activeProxy to null
+    if (enableProxy && proxyState.proxies.length > 0) {
+      activeProxy = proxyState.proxies[0]
+    }
+    updateProxyState({
+      enabled: enableProxy,
+      activeProxy,
+    })
+  }
+
+  const addProxy = async (proxyUrl: string) => {
+    if (proxyState.proxies.includes(proxyUrl)) {
+      log.warn('skip already existing proxy', proxyUrl)
+      // proxy alread exists, should we show a message here?
+      return
+    }
```

Comment:
```suggestion
    if (proxyState.proxies.includes(proxyUrl)) {
      log.warn('skip already existing proxy', proxyUrl)
      setShowNewProxyForm(false)
      setNewProxyUrl('')
      // proxy alread exists, should we show a message here?
      return
    }
```

Reply:
Not sure if that is expected behaviour. It would look like as if it worked. I think it would be better to disable the button in that case. Just tested and seems we need some new disabled colors for primary & secondary buttons...

Reply:
Well, arguably it did work.

Otherwise I'd vote for showing an error message.

Reply:
Button is disabled now